From e84f00c45ad064f1503d2eb03d7600e59744f0e6 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 29 Sep 2020 15:03:34 +0100
Subject: [PATCH] ASoC: cs42xx8: Only define cs42xx8_of_match once

cs42xx8.c exports cs42xx8_of_match, so there's no need to redefine it
in cs42xx8-i2c.c - doing so breaks linking when loadable module
support is disabled. It would be tidy to use the exported match table
in cs42xx8.c's of_match_table member, but an imported symbol can't be
used in a module's MODULE_DEVICE_TABLE declaration. Instead, rename
the duplicated declarations so as not to clash.

See: https://github.com/raspberrypi/linux/issues/3873

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 sound/soc/codecs/cs42xx8-i2c.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/sound/soc/codecs/cs42xx8-i2c.c
+++ b/sound/soc/codecs/cs42xx8-i2c.c
@@ -45,18 +45,18 @@ static struct i2c_device_id cs42xx8_i2c_
 };
 MODULE_DEVICE_TABLE(i2c, cs42xx8_i2c_id);
 
-const struct of_device_id cs42xx8_of_match[] = {
+const struct of_device_id cs42xx8_i2c_of_match[] = {
 	{ .compatible = "cirrus,cs42448", .data = &cs42448_data, },
 	{ .compatible = "cirrus,cs42888", .data = &cs42888_data, },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(of, cs42xx8_of_match);
+MODULE_DEVICE_TABLE(of, cs42xx8_i2c_of_match);
 
 static struct i2c_driver cs42xx8_i2c_driver = {
 	.driver = {
 		.name = "cs42xx8",
 		.pm = &cs42xx8_pm,
-		.of_match_table = cs42xx8_of_match,
+		.of_match_table = cs42xx8_i2c_of_match,
 	},
 	.probe = cs42xx8_i2c_probe,
 	.remove = cs42xx8_i2c_remove,
